kie-spring/src/main/java/org/kie/spring/manager/SpringRuntimeManagerFactoryImpl.java

```diff
@@ -30,19 +30,20 @@
 import org.springframework.transaction.support.AbstractPlatformTransactionManager;
 
 public class SpringRuntimeManagerFactoryImpl extends RuntimeManagerFactoryImpl {
-	
+
 	private AbstractPlatformTransactionManager transactionManager;
-	
 	private UserGroupCallback userGroupCallback;
-
     private EntityManager entityManager;
-
     private UserInfo userInfo;
-
     private boolean pessimisticLocking = false;
-
     private GlobalSchedulerService schedulerService;
+    private static final String TX_FACTORY_CLASS = "org.kie.txm.factory.class";
+    private static final String SPRING_TX_FACTORY_CLASS = "org.kie.spring.persistence.KieSpringTransactionManagerFactory";
 
+    public SpringRuntimeManagerFactoryImpl() {
+        System.setProperty(TX_FACTORY_CLASS, SPRING_TX_FACTORY_CLASS);
```

Comment:
I'm curious, why don't you just use?

```suggestion
        System.setProperty("org.kie.txm.factory.class", KieSpringTransactionManagerFactory.class.getCanonicalName());
```
and remove L40 and L41 since are not used again :thinking: 
(or move `KieSpringTransactionManagerFactory.class.getCanonicalName()` to static instance var in case of performance issues)

this way you avoid problems in case `KieSpringTransactionManagerFactory` is refactored.
Sorry in case the question is so stupid, probably is a lack of context from my side

Reply:
Well, I don't think it makes too much difference, I agree that if the class is named, you may not be aware of the change inmmediately, and skip that refactor.
In the other hand, by using `KieSpringTransactionManagerFactory.class.getCanonicalName()` we are creating an additional dependency in that particular class.

So IMHO, I think both approaches are ok... but just for consistency with the [rest of the code](https://github.com/kiegroup/droolsjbpm-integration/blob/main/kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/autoconfigure/JBPMAutoConfiguration.java#L138-L139), I'd prefer to keep it as is.